src/core/EventBus.js

```diff
@@ -49,11 +51,13 @@ function EventBus() {
 
         var handler = {
             callback: listener,
-            scope: scope
+            scope: scope,
+            priority: priority !== undefined ? priority : LOW_PRIORITY
         };
 
         handlers[type] = handlers[type] || [];
         handlers[type].push(handler);
+        handlers[type].sort(function (a, b) {return a.priority - b.priority;});
```

Comment:
I disagree with you on this one.  Event priority is part of almost every event system I have ever worked with such as Signals etc.  I think we can make it more useful and much more clear but only allowing order to be determined based on event registration seems very limiting.   I do agree that we should first look for a refactor,  in this case I did but it was not trivial at all.  Chicken/ Egg issue.